analyzers/src/SonarAnalyzer.CSharp/Rules/InfiniteRecursion.RoslynCfg.cs

```diff
@@ -0,0 +1,151 @@
+                            return;
+                        }
+                    }
+
+                    cfg = cfg.GetLocalFunctionControlFlowGraph(symbol as IMethodSymbol);
+                    if (cfg == null)
+                    {
+                        return;
+                    }
+                }
+
+                var context = new RecursionContext<ControlFlowGraph>(cfg, symbol, identifier.GetLocation(), c, "method's recursion");
+                var walker = new RecursionSearcher(context);
+                walker.CheckPaths();
+            }
+
+            private class RecursionSearcher : CfgAllPathValidator
+            {
+                private readonly RecursionContext<ControlFlowGraph> context;
+                private readonly bool isGetAccesor;
+
+                public RecursionSearcher(RecursionContext<ControlFlowGraph> context, bool isGetAccesor = true)
+                    : base(context.ControlFlowGraph)
+                {
+                    this.context = context;
+                    this.isGetAccesor = isGetAccesor;
+                }
+
+                public void CheckPaths()
+                {
+                    if (CheckAllPaths() || CfgHasUnescapableLoop())
+                    {
+                        context.ReportIssue();
+                    }
+                }
+
+                protected override bool IsValid(BasicBlock block)
+                {
+                    foreach (var operation in block.OperationsAndBranchValue)
+                    {
+                        foreach (var child in operation.ToReversedExecutionOrder())
+                        {
+                            if (context.AnalyzedSymbol.Equals(MemberSymbol(child.Instance)))
+                            {
+                                var isWrite = child.Parent is { Kind: OperationKindEx.SimpleAssignment } parent && ISimpleAssignmentOperationWrapper.FromOperation(parent).Target == child.Instance;
+                                return isGetAccesor ^ isWrite;
+                            }
+                        }
+                    }
+                    return false;
+
+                    static ISymbol MemberSymbol(IOperation operation) =>
+                        operation.Kind switch
+                        {
+                            OperationKindEx.PropertyReference => IPropertyReferenceOperationWrapper.FromOperation(operation).Property,
+                            OperationKindEx.Invocation => IInvocationOperationWrapper.FromOperation(operation).TargetMethod,
+                            _ => null
+                        };
+                }
+
+                protected override bool IsInvalid(BasicBlock block) => false;
+
+                private bool CfgHasUnescapableLoop() =>
+                    !context.ControlFlowGraph.ExitBlock.IsReachable
+                    && !context.ControlFlowGraph.Blocks.Any(x => x.FallThroughSuccessor?.Semantics == ControlFlowBranchSemantics.Throw && x.IsReachable);
```

Comment:
`Un*` + `!` + `!` = too many negations. Let's make it positive. Something like `CfgCanExit`